verifier/src/evidence.rs

```diff
@@ -0,0 +1,334 @@
+// Copyright (c) 2023 The MobileCoin Foundation
+
+//! The full set of evidence needed for attesting a quote
+
+use crate::{Accessor, Advisories, Error, Result, TcbInfo, TcbInfoRaw};
+use der::DecodePem;
+use mc_sgx_core_types::{
+    Attributes, ConfigId, ConfigSvn, CpuSvn, ExtendedProductId, FamilyId, IsvProductId, IsvSvn,
+    MiscellaneousSelect, MrEnclave, MrSigner, ReportData,
+};
+use mc_sgx_dcap_types::{CertificationData, Quote3, TcbInfo as QuoteTcbInfo};
+use x509_cert::Certificate;
+
+/// The full set of evidence needed for verifying a quote
+///
+/// A wrapping container for a `Quote3` and a `TcbInfoRaw`. This can be used
+/// with the majority of the `Verifier` implementations from this crate.
+/// This allows one to compose one verifier and use an [`Evidence`] instance in
+/// the `verify()` method.
+///
+/// Importantly this will derive the [`Advisories`] related to the provided
+/// `quote` and `tcb_info_raw`, so that one can verify the allowed advisories.
+#[derive(Debug)]
+pub struct Evidence<'a, Q> {
+    quote: Quote3<Q>,
+    tcb_info_raw: TcbInfoRaw<'a>,
+    advisories: Advisories,
+}
+
+impl<'a, Q: AsRef<[u8]>> Evidence<'a, Q> {
+    /// Create a new instance
+    pub fn new(quote: Quote3<Q>, tcb_info_raw: TcbInfoRaw<'a>) -> Result<Self> {
+        let quote_tcb_info = tcb_info_try_from_quote(&quote)?;
+        let tcb_info = TcbInfo::try_from(&tcb_info_raw)?;
+        let advisories = tcb_info.advisories(&quote_tcb_info)?;
+        Ok(Self {
+            quote,
+            tcb_info_raw,
+            advisories,
+        })
+    }
+}
+
+impl<'a, Q> Accessor<TcbInfoRaw<'a>> for Evidence<'a, Q> {
+    fn get(&self) -> TcbInfoRaw<'a> {
+        self.tcb_info_raw.clone()
+    }
+}
+
+impl<'a, Q: Clone> Accessor<Quote3<Q>> for Evidence<'a, Q> {
+    fn get(&self) -> Quote3<Q> {
+        self.quote.clone()
+    }
+}
+
+impl<'a, Q> Accessor<Advisories> for Evidence<'a, Q> {
+    fn get(&self) -> Advisories {
+        self.advisories.clone()
+    }
+}
+
+// TODO think this should go in tcb.rs of `mc-sgx-dcap-types`
+fn tcb_info_try_from_quote<Q: AsRef<[u8]>>(quote: &Quote3<Q>) -> Result<QuoteTcbInfo> {
```

Comment:
Like the TODO says IFF this overall PR is the path we want to go then I think we should consider adding a 
```rust
impl TryFrom<Quote3> for TcbInfo{
```
in the `mc-sgx-dcap-types`